app/services/transaction_service/process/preauthorize.rb

```diff
@@ -8,12 +10,28 @@ class Preauthorize
     def create(tx:, gateway_fields:, gateway_adapter:, prefer_async:)
       Transition.transition_to(tx[:id], :initiated)
 
-      Gateway.unwrap_completion(
-        gateway_adapter.create_payment(
-          tx: tx,
-          gateway_fields: gateway_fields,
-          prefer_async: prefer_async)) do
+      if use_async?(prefer_async, gateway_adapter)
+        proc_token = Worker.enqueue_preauthorize_op(
+          community_id: tx[:community_id],
+          transaction_id: tx[:id],
+          op_name: :do_create,
+          op_input: [tx, gateway_fields])
 
+        proc_status_response(proc_token)
+      else
+        do_create(tx, gateway_fields)
+      end
+    end
+
+    def do_create(tx, gateway_fields)
+      gateway_adapter = TransactionService::Transaction.gateway_adapter(tx[:payment_gateway])
+
+      completion = gateway_adapter.create_payment(
+        tx: tx,
+        gateway_fields: gateway_fields,
+        prefer_async: false)
```

Comment:
We now have an option prefer_async which is named so that it hints what's the preferred behavior but cannot fully control it. Does this still make sense? When we're already in a worker context it feels to me we should be able to just force the adapter to go sync so don't end up scheduling a job from a job. Also, we wouldn't have to deal with the async completion case at all then. prefer_async already works prescriptively so it's almost purely about naming and communicating the intent correctly (to force and not hint).